### PR TITLE
Added the proper logging workflow in upgrade

### PR DESCRIPTION
**Pull request by devendra104**
#379

### CODE REVIEW

**Comment by devendra104** [comment]
> Thanks for taking this up @devendra104, its long pending improvement.
> 
> Btw loggings need to be more precise and clear on what entity the change is happeing / going to happen. The names are better than ids and clear. If you want ids for debugging purpose keep both name and id in logging.
> 
> Also pre logging before major happenings like publish, sync is very welcome as a person get an idea on whats happening untill the success message.

Thanks @jyejare for providing the valuable comment.

File: upgrade/helpers/tasks.py [hunk 1]
```diff
@@ -113,21 +113,30 @@ def sync_capsule_repos_to_upgrade(capsules):
...
+    logger.info("Activation key {} used for capsule subscription has found".
+                format(ak_name))
     cv = ak.content_view.read()
     lenv = ak.environment.read()
     # Fix dead pulp tasks
     if os_ver == '6':
         run('for i in pulp_resource_manager pulp_workers pulp_celerybeat; '
             'do service $i restart; done')
     _sync_capsule_subscription_to_capsule_ak(ak)
+    logger.info("Capsule subscription to AK {} has added successfully".format(ak))
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]: ak is an entire object, you should use ak.name ??

File: upgrade/helpers/tasks.py [hunk 2]
```diff
@@ -113,21 +113,30 @@ def sync_capsule_repos_to_upgrade(capsules):
...
     if float(to_version) >= 6.3:
         _add_additional_subscription_for_capsule(ak, capsuletools_url)
     # Publishing and promoting the CV with all newly added capsule, capsuletools, rhscl and
     # server repos combine
+    logger.info("Content view publish operation has started successfully")
     call_entity_method_with_timeout(cv.read().publish, timeout=2000)
+    logger.info("Content view publish operation has completed successfully")
     published_ver = entities.ContentViewVersion(
         id=max([cv_ver.id for cv_ver in cv.read().version])).read()
+    logger.info("Published version {} promotion has started successfully".
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]: same here its an object, use either a version or cv name.

File: upgrade/helpers/tasks.py [hunk 3]
```diff
@@ -113,21 +113,30 @@ def sync_capsule_repos_to_upgrade(capsules):
...
+                format(published_ver))
     published_ver.promote(data={'environment_id': lenv.id, 'force': False})
+    logger.info("Published version {} promotion has completed successfully".
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]: same as above.

File: upgrade/helpers/tasks.py [hunk 4]
```diff
@@ -187,6 +210,9 @@ def _sync_capsule_subscription_to_capsule_ak(ak):
             'quantity': 1,
             'subscription_id': cap_sub.id,
         })
+        logger.info("subscription {} in Activation key used for Capsule subscription "
+                    "added successfully for subscription id: {}".format(cap_sub.id,
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]:
```suggestion
                    "added successfully for subscription: {}".format(cap_sub.id,
```

File: upgrade/helpers/tasks.py [hunk 5]
```diff
@@ -211,12 +240,17 @@ def _sync_rh_repos_to_satellite(org):
     scl_product = entities.Product(
         name=rhelcontents['rhscl_sat64']['prod'], organization=org
     ).search(query={'per_page': 100})[0]
+    logger.info("Red Hat Software Collection for product {} is enabled and found".
+                format(rhelcontents['rhscl_sat64']['prod']))
     scl_reposet = entities.RepositorySet(
         name=rhelcontents['rhscl']['repo'].format(os_ver=rhelver), product=scl_product
     ).search()[0]
+    logger.info("Red Hat Software Collection forrepos {} is already enabled and found".
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]:
```suggestion
    logger.info("Red Hat Software Collection for repos {} is already enabled and found".
```

File: upgrade/helpers/tasks.py [hunk 6]
```diff
@@ -237,20 +271,29 @@ def _sync_rh_repos_to_satellite(org):
...
+                format(rhelcontents['server']['prod']))
     server_reposet = entities.RepositorySet(
         name=rhelcontents['server']['repo'].format(os_ver=rhelver), product=server_product
     ).search()[0]
+    logger.info("Repository {} is already enabled and found".
+                format(rhelcontents['server']['repo']))
     try:
         server_reposet.enable(
             data={'basearch': arch, 'releasever': '7Server', 'organization_id': org.id})
+        logger.info("Repository enabled successfully for base search {}, 7Serve".format(arch))
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]:
```suggestion
        logger.info("Repository enabled successfully for base arch {}, 7Server".format(arch))
```

File: upgrade/helpers/tasks.py [hunk 7]
```diff
@@ -274,23 +317,39 @@ def _sync_sattools_repos_to_satellite_for_capsule(capsuletools_url, org):
             name=customcontents['capsule_tools']['repo'],
             product=captools_product, url=capsuletools_url, organization=org, content_type='yum'
         ).create()
+        logger.info("The custom tools product {} and repository {} is created from "
+                    "capsule tools url".format(customcontents['capsule_tools']['repo'],
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]: Format is interchanged here, use prod for prod and repo for repo.

File: upgrade/helpers/tasks.py [hunk 8]
```diff
@@ -274,23 +317,39 @@ def _sync_sattools_repos_to_satellite_for_capsule(capsuletools_url, org):
...
         time.sleep(5)
         captools_repo = entities.Repository(
             name=rhelcontents['tools']['repofull'].format(
                 sat_ver=to_ver, os_ver=rhelver, arch=arch)
         ).search(query={'organization_id': org.id, 'per_page': 100})[0]
-    call_entity_method_with_timeout(entities.Repository(id=captools_repo.id).sync, timeout=2500)
+        logger.info("Entities repository search completed successfully for tools "
+                    "repo {}".format(rhelcontents['tools']['repofull'].
+                                     format(sat_ver=to_ver, os_ver=rhelver, arch=arch)))
+    logger.info("Entities repository sync completed successfully for capsule repo name{}".
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]: s/completed/started

File: upgrade/helpers/tasks.py [hunk 9]
```diff
@@ -323,13 +388,17 @@ def _add_additional_subscription_for_capsule(ak, capsuletools_url):
...
+        logger.info("Activation key successfully override for capsule content_label {}".
+                    format(captools_repo.repo_name))
     else:
         captools_sub = entities.Subscription().search(
             query={'search': 'name={0}'.format(customcontents['capsule_tools']['prod'])})[0]
         ak.add_subscriptions(data={
             'quantity': 1,
             'subscription_id': captools_sub.id,
         })
+        logger.info("Capsule Tools subscription {} added successfullyto capsule AK".
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]:
```suggestion
        logger.info("Capsule Tools subscription {} added successfully to capsule AK".
```

File: upgrade/helpers/tasks.py [hunk 10]
```diff
@@ -386,19 +455,29 @@ def sync_tools_repos_to_upgrade(client_os, hosts):
...
+    logger.info("Entities product {} and repository {} is created successfully".
+                format(tools_product, toolsrepo_name))
     entities.Repository(id=tools_repo.id).sync()
+    logger.info("Entities repository sync operation has completed successfully for tool "
+                "repos name {}".format(tools_repo.name))
     cv.repository += [tools_repo]
     cv.update(['repository'])
+    logger.info("Content view publish operation is started successfully")
     call_entity_method_with_timeout(cv.read().publish, timeout=2500)
+    logger.info("Content view has published successfully within in 2500 seconds")
```
Comment (jyejare) [inline upgrade/helpers/tasks.py]:
```suggestion
    logger.info("Content view has published successfully")
```
You dont whether it really took 2500 seconds or 1000 seconds.

**Pull request merged**